src/dmd/expression.d

```diff
@@ -4285,22 +4285,22 @@ extern (C++) final class VarExp : SymbolExp
     {
         if (var.storage_class & STC.manifest)
         {
-            error("manifest constant `%s` is not lvalue", var.toChars());
+            error("manifest constant `%s` cannot be modified", var.toChars());
             return new ErrorExp();
         }
         if (var.storage_class & STC.lazy_)
         {
-            error("lazy variables cannot be lvalues");
+            error("lazy variables cannot be modified");
```

Comment:
While you're modifying the line anyway, how about upgrading it to:
```
error("lazy variable `%s` cannot be modified", var.toChars());
```

Reply:
Good idea! Thanks! I also added a test case for it as it was uncovered.